### PR TITLE
DRY-up the MetadataRequest_v1 struct

### PR DESCRIPTION
**Pull request by jeffwidman**

### CODE REVIEW

File: kafka/protocol/metadata.py [hunk 1]
```diff
@@ -52,17 +52,15 @@ class MetadataRequest_v0(Struct):
     API_VERSION = 0
     RESPONSE_TYPE = MetadataResponse_v0
     SCHEMA = Schema(
-        ('topics', Array(String('utf-8'))) # Empty Array (len 0) for all topics
+        ('topics', Array(String('utf-8')))  # Empty Array (len 0) for all topics
```
Comment (jeffwidman) [inline kafka/protocol/metadata.py]: Should this have additional class vars?

    ALL_TOPICS = None  # Empty Array (len 0) for all topics

File: kafka/protocol/metadata.py [hunk 2]
```diff
@@ -52,17 +52,15 @@ class MetadataRequest_v0(Struct):
...
 
 
 class MetadataRequest_v1(Struct):
     API_KEY = 3
     API_VERSION = 1
     RESPONSE_TYPE = MetadataResponse_v1
-    SCHEMA = Schema(
-        ('topics', Array(String('utf-8'))) # Null Array (len -1) for all topics
-    )
+    SCHEMA = MetadataRequest_v0.SCHEMA  # Null Array (len -1) for all topics
```
Comment (jeffwidman) [inline kafka/protocol/metadata.py]: Should this have additional class vars?

    ALL_TOPICS = -1  # Null Array (len -1) for all topics
    NO_TOPICS = None

File: kafka/protocol/metadata.py [hunk 3]
```diff
@@ -52,17 +52,15 @@ class MetadataRequest_v0(Struct):
     API_VERSION = 0
     RESPONSE_TYPE = MetadataResponse_v0
     SCHEMA = Schema(
-        ('topics', Array(String('utf-8'))) # Empty Array (len 0) for all topics
+        ('topics', Array(String('utf-8')))  # Empty Array (len 0) returns all topics
```
Comment (jeffwidman) [inline kafka/protocol/metadata.py]: Should this have additional class vars?

    ALL_TOPICS = None  # Empty Array (len 0) returns all topics

Not sure what correct values are for empty arrays in this structs class

File: kafka/protocol/metadata.py [hunk 4]
```diff
@@ -52,17 +52,15 @@ class MetadataRequest_v0(Struct):
...
 
 
 class MetadataRequest_v1(Struct):
     API_KEY = 3
     API_VERSION = 1
     RESPONSE_TYPE = MetadataResponse_v1
-    SCHEMA = Schema(
-        ('topics', Array(String('utf-8'))) # Null Array (len -1) for all topics
-    )
+    SCHEMA = MetadataRequest_v0.SCHEMA  # Null Array (len -1) returns all topics
```
Comment (jeffwidman) [inline kafka/protocol/metadata.py]: Should this have additional class vars?

    ALL_TOPICS = -1  # Null Array (len -1) returns all topics
    NO_TOPICS = None  # Empty array (len 0) returns no topics

Not sure what correct values are for empty arrays in this structs class

Comment (dpkp) [inline kafka/protocol/metadata.py]: Sure, we could add something like that to make it easier to remember which is which

Comment (jeffwidman) [inline kafka/protocol/metadata.py]: Pushed an updated version that includes these class vars. I'm still not sure what correct values are for empty arrays in this structs class, so I set them to `None`, but perhaps they should be `0`.

**Pull request merged**